### PR TITLE
dark theme: update background colors

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksupdate-background-color-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1203652232454021/1204107070515701) 🎟️

## 🗒️ What

Updates instances of `background-color: white` with `background-color: var(--token-color-surface-primary)` ([which is #fff](https://helios.hashicorp.design/foundations/colors?tab=palette#surface-1) in light mode) 

Also makes an override for the card component background color. 

### Affected Components:
- Card
- MobileMenuContainer
- SidebarNavMenuItem
- TabButtonControls
- TabDropdownControls
- VersionContextSwitcher dropdown
- VideoEmbed
- Integrations VariableGroupList item
- Dialog

## 🧪 Testing

- [Visit the preview](https://dev-portal-git-ksupdate-background-color-hashicorp.vercel.app/), choose dark mode

To test the following components, switch between light and dark. Light mode should match production.
- Visit a [sidebar sidecar layout page,](https://dev-portal-git-ksupdate-background-color-hashicorp.vercel.app/vault) view the cards, sidebar, main content area background, hit the 'bookmark' icon to see the dialog with the dark background
- [Checkout install view](https://dev-portal-git-ksupdate-background-color-hashicorp.vercel.app/vault/downloads), look at the tab button controls, version switcher dropdown
- Go to [integrations component view page](https://dev-portal-git-ksupdate-background-color-hashicorp.vercel.app/waypoint/integrations/hashicorp/aws-ec2/latest/components/platform), checkout the group list item background
- Visit a[ tutorial with a video embed](https://dev-portal-git-ksupdate-background-color-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-intro)

## 💭 Anything else?

This isn't the final state necessarily, but a first step to move away from manual color definitions and fix-up fully _broken_ background colors.

### CODE REVIEW

File: src/layouts/core-dev-dot-layout/core-dev-dot-layout.module.css [hunk 1]
```diff
@@ -32,6 +32,7 @@
   font-family: var(--font-body) we set on our `body` element needs to be
   re-declared here, so that our "system-ui" font stack cascades as expected. */
 	font-family: var(--font-body);
+	background-color: var(--token-color-surface-primary);
```
Comment (kendallstrautman) [inline src/layouts/core-dev-dot-layout/core-dev-dot-layout.module.css:35]: Adding a background color to the entire layout, lmk if folks disagree here. This way we don't have to go in and set background color on the sidebar and main content areas. But i'm open for that approach also

File: src/components/card/card.module.css [hunk 2]
```diff
@@ -7,8 +7,12 @@
 	/* used by components like CardLink */
 	--card-border-radius: 6px;
 
-	background-color: white;
+	background-color: var(--token-color-surface-primary);
```
Comment (kendallstrautman) [inline src/components/card/card.module.css:10]: In 'light' mode, [this is the same as white or #fff](https://helios.hashicorp.design/foundations/colors?tab=palette#surface-1)

File: src/components/card/card.module.css [hunk 3]
```diff
@@ -7,8 +7,12 @@
...
 	border-radius: var(--card-border-radius);
 	padding: 16px;
 	width: 100%;
+
+	@nest html[data-theme='dark'] & {
+		background-color: var(--token-color-surface-faint);
+	}
```
Comment (kendallstrautman) [inline src/components/card/card.module.css:17]: First override here 🤝 Open to feedback on this approach.

File: src/layouts/core-dev-dot-layout/core-dev-dot-layout.module.css (hunk 1 above)
Comment (zchsh) [reply src/layouts/core-dev-dot-layout/core-dev-dot-layout.module.css:35]: I dig the approach you have here 🙌

**Review by zchsh — APPROVED** [review]

**Comment by zchsh** [comment]
⛏️ Oooh this is a total nitpick, it seems like things actually look _fine_ even without this change, but this could be a good opportunity to set up a global CSS var for `var(--token-color-surface-primary-transparent)`, for sidecar 🤔

https://github.com/hashicorp/dev-portal/blob/391883c4c2bb4059f0fe00aeb3ba9ea7f25b8e98/src/layouts/sidebar-sidecar/components/sidecar-scroll-container/sidecar-scroll-container.module.css#L10-L11

That being said, it seems to look fine without this change, so certainly not blocking (maybe better to wait for upstream HDS support for this, and not worry about it for now, i think that'd be fine 👍)

| Gradient at normal opacity | Gradient at fully opaque to show color |
| - | - |
| CleanShot 2023-03-07 at 10 29 49@2x | CleanShot 2023-03-07 at 10 27 50@2x |

**Comment by kendallstrautman** [comment]
@zchsh great callout! I will lift this up to a global token

**Pull request merged**